crates/re_types_builder/src/context.rs

```diff
@@ -0,0 +1,125 @@
+use std::sync::mpsc;
+
+/// Creates a new context.
+///
+/// The [`Context`] can be freely cloned and sent to other threads.
+///
+/// The [`ContextRoot`] should not be sent to other threads.
+pub fn context() -> (ContextRoot, Context) {
+    let (tx, rx) = mpsc::channel();
+    (ContextRoot::new(rx), Context::new(tx))
+}
+
+#[derive(Clone)]
+pub struct Context {
```

Comment:
`Context` is very non-descript. I suggest `ErrorContext`.

Reply:
Maybe `Reporter`?

Consider this:

```
    fn generate(
        &mut self,
        ctx: &Context,
        objects: &Objects,
        arrow_registry: &ArrowRegistry,
    )
```

`Context` here speaks very little about what it is for. Besides, `ArrowRegistry` is already context for the function (its job is to generate code for the objects, and using the registry as context/environment)

Reply:
Renamed it to `Reporter`, and `ContextRoot` to `Report`